cmake/modules/AddSwiftUnittests.cmake

```
@@ -42,7 +42,7 @@ function(add_swift_unittest test_dirname)
   if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
     # Add an @rpath to the swift library directory.
```

(Optional) Would it be worth to update this with the new scenario? Something akin to

```suggestion
    # Add an @rpath to the swift library directory
    # and one to the OS dylibs we require but
    # are not building ourselves (e.g Foundation overlay)
```